bots/reads.ts

```diff
@@ -410,36 +412,42 @@ const handleUpdateUrl = async (url: string, ctx: ReadsContext, config: ReadsConf
   ctx.session.item.link = cleanUrl;
 
   try {
-    metadata = await fetchUrlMetadata(cleanUrl, config);
-    await ensureNonDuplicateLink(cleanUrl, config);
+      metadata = await fetchUrlMetadata(cleanUrl, config);
+      await ensureNonDuplicateLink(cleanUrl, config);
+
+      const { description, image, title } = metadata
+      
+      ctx.session.item.title = title || '';
+      ctx.session.item.tags = defaultTagsForUrl(cleanUrl);
+
+      ctx.session.item.description
+        = (description && description.length > 500)
+          ? description.substring(0, 497) + '...'
+          : description || '';
+    
+      ctx.session.item.image_url = image || '';
   } catch (e) {
-    console.error('Error proccessing handleUpdateUrl: ', e);
     if (e.message === ERROR_DUPLICATE_READ) {
       await ctx.reply('Oops, this url was recently added already');
       resetState(ctx);
       return;
     }
 
-    await ctx.reply('sorry, I could not fetch that url');
-    await handleNew(ctx);
-    return;
-  }
-
-  const { description, image, title } = metadata;
-
-  if (!cleanUrl.includes('x.com')) {
-    // not twitter, so save the title
-    ctx.session.item.title = title || '';
+    try {
+      const summary = await fetchUrlSummary(cleanUrl, config.openaiApi)
```

Comment:
the way this is setup in this PR, the ai generation call will only happen if the above metadata call fails and it would happen for all urls including `x.com`. per the instructions this call should happen for all non-`x.com` urls, and it shouldn't be dependent on the metadata call failing. 

please restructure this code the way it is in the main branch and only call the `fetchUrlSummary` function if the url is not an `x.com` url. the structure for this was already setup to handle this well and just needs the new call injected there and the description overwritten if one comes back from OpenAI. see:

https://github.com/delphidigital/delphi-tg-bots/blob/main/bots/reads.ts#L435-L438